File: Dia61/CalculaMedia.js
// Calcula la Media en JavaScript

function media(...numeros) {
  if (typeof numeros[0] !== 'number') {
    numeros = numeros.flat()
  }
  const suma = numeros.reduce((a, b) => a + b)
  const n = numeros.length
  return suma / n
}

console.log(media(1, 2, 3, 4)) // 2.5
console.log(media([1, 2, 3, 4])) // 2.5

File: Dia61/CalculaMedia.py
# Calcula la Media en Python

def esNumero(x):
  try:
    float(x)
    return True
  except:
    return False

def media(*numeros):
  if (esNumero(numeros[0]) == False):
    numeros = numeros[0]
  else:
    numeros = list(numeros)
  suma = 0
  for n in numeros: suma += n
  n = len(numeros)
  return suma / n

print(media(1, 2, 3, 4)) # 2.5
print(media([1, 2, 3, 4])) # 2.5